core/src/LegOdometry.hpp

```diff
@@ -29,58 +30,114 @@
     void computeIMP(const std::string& frame, const Eigen::Matrix3d& R,
                     const Eigen::Vector3d& angular_velocity, const Eigen::Vector3d& linear_velocity,
                     const Eigen::Vector3d& force, std::optional<Eigen::Vector3d> torque = std::nullopt);
 
-    bool is_initialized{};
-    Eigen::Vector3d base_position_ = Eigen::Vector3d::Zero();
-    Eigen::Vector3d base_position_prev_ = Eigen::Vector3d::Zero();
-    Eigen::Vector3d base_linear_velocity_ = Eigen::Vector3d::Zero();
-    Params params_;
+    bool is_initialized{}; ///< Flag indicating if the odometry has been initialized
+    Eigen::Vector3d base_position_ = Eigen::Vector3d::Zero(); ///< Estimated base position in world coordinates
+    Eigen::Vector3d base_position_prev_ = Eigen::Vector3d::Zero(); ///< Previous base position in world coordinates
+    Eigen::Vector3d base_linear_velocity_ = Eigen::Vector3d::Zero(); ///< Estimated base linear velocity in world coordinates
+    Params params_; ///< Optimization parameters
 
-    std::map<std::string, Eigen::Vector3d> pivots_;
-    std::map<std::string, Eigen::Vector3d> contact_positions_;
-    std::map<std::string, Eigen::Quaterniond> contact_orientations_;
+    std::map<std::string, Eigen::Vector3d> pivots_; ///< Pivot points for the feet in relative foot coordinates
+    std::map<std::string, Eigen::Vector3d> contact_positions_; ///< Contact positions of the feet relative to the base frame
+    std::map<std::string, Eigen::Quaterniond> contact_orientations_; ///< Contact orientations of the feet relative to the base frame
 
-    std::map<std::string, Eigen::Vector3d> feet_position_prev_;
-    std::map<std::string, Eigen::Quaterniond> feet_orientation_prev_;
+    std::map<std::string, Eigen::Vector3d> feet_position_prev_; ///< Previous positions of the feet in world coordinates
+    std::map<std::string, Eigen::Quaterniond> feet_orientation_prev_; ///< Previous orientations of the feet in world coordinates
 
-    std::optional<std::map<std::string, Eigen::Vector3d>> force_torque_offset_;
+    std::optional<std::map<std::string, Eigen::Vector3d>> force_torque_offset_; ///< Force/Torque sensor offsets from the contact frames
 
    public:
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
+    /**
+     * @brief Constructor for the LegOdometry class.
+     * @param feet_position Initial positions of the feet in world coordinates
+     * @param feet_orientation Initial orientations of the feet in world coordinates
+     * @param mass Mass of the robot
+     * @param alpha1 Alpha1 parameter
+     * @param alpha3 Alpha3 parameter
+     * @param freq Joint state nominal frequency
+     * @param g Gravity constant
+     * @param eps Epsilon parameter
+     * @param force_torque_offset Force/Torque sensor offsets from the contact frames (optional)
+     */
     LegOdometry(
         std::map<std::string, Eigen::Vector3d> feet_position,
         std::map<std::string, Eigen::Quaterniond> feet_orientation, double mass = 5.14,
```

Comment:
Maybe remove default mass value and add a check if it has value because it is not optional

Reply:
removed default values